### ISSUE TITLE
Fragment merging excluding fields

### ISSUE THREAD

**Issue by peterhorne** [rmosolgo/graphql-ruby#86]
Similar to #55 I reported a little while ago. This time I am issuing the following query:

```
query Query {
  viewer {
    id
    payment(id: "f7781868-841c-4a32-94e9-8921a73635f7") {
      id
      ...__PaymentFragment1
      ...__PaymentFragment2
    }
  }
}

fragment __PaymentFragment1 on Payment {
  payee {
    name
    currency
  }
}

fragment __PaymentFragment2 on Payment {
  payee {
    name
  }
}
```

Whatever is in `__PaymentFragment1` is completely ignored and so `currency` is never returned. Moving `currency` into `__PaymentFragment2` will cause it to be returned in the response.

**Comment by peterhorne** [comment]
I see this is a dupe of https://github.com/rmosolgo/graphql-ruby/pull/85. Have tried against master and it fixes the problem... yay!

**Comment by rmosolgo** [comment]
Sorry 'bout that, just shipped it in 0.10.9!
